server/src/test/java/org/elasticsearch/search/aggregations/metrics/MaxIT.java

```diff
@@ -411,4 +412,34 @@ public void testEarlyTermination() throws Exception {
         assertThat(count.getName(), equalTo("count"));
         assertThat(count.getValue(), equalTo(20L));
     }
+
+    public void testNestedEarlyTermination() throws Exception {
+        for (Aggregator.SubAggCollectionMode collectionMode : Aggregator.SubAggCollectionMode.values()) {
+            SearchResponse searchResponse = client().prepareSearch("idx")
+                .setTrackTotalHits(false)
+                .setQuery(matchAllQuery())
+                .addAggregation(max("max").field("values"))
+                .addAggregation(count("count").field("values"))
```

Comment:
An unrelated question: is the `count` agg required here to check for correctness?  It looks like `count` doesn't use the early termination optimization, so wasn't sure if that is being used as a proxy to make sure that other aggs are unaffected, or something?

Asking because @csoulios is converting the max IT to unit tests, and we were discussing the other early termination test last week. I wasn't quite sure the purpose of `count` here, or if it was included because why not :)

Reply:
Yes, it is used to check if a sibling is not affected by an aggregation that terminates early. So the fact that we don't collect any hits for the `max` aggregation should not impact the `count` one which needs to see all hits.

Reply:
Gotcha, thanks.  So if we embedded this inside a `filter` agg with `match_all` it should be functionally the same, right?

`AggregatorTestCase` currently assumes a single top-level agg for simplicity, so when converting these IT over to unit tests we weren't sure how to handle it.  We could also just leave the early termination tests as IT too.

(Note: not for this PR, just discussion regarding the changes we're making to tests in a separate PR)

Reply:
> So if we embedded this inside a filter agg with match_all it should be functionally the same, right?

Yes